pkg/parquetconverter/parquet_converter.go

```diff
@@ -259,40 +288,58 @@ func (c *ParquetConverter) running(ctx context.Context) error {
 						continue
 					}
 
-					mark, err := ReadConversionMark(ctx, m.ULID, uBucket, ulogger)
-					if err != nil {
-						level.Error(ulogger).Log("msg", "failed to read conversion mark, skipping", "err", err, "block", m.ULID.String())
-						continue
-					}
+					c.processBlock(ctx, u, m, uBucket, ulogger)
+				}
+			}
+		}
+	}
+}
 
-					if mark.Version == CurrentVersion {
-						continue
-					}
+// processBlock handles the conversion of a single block with proper metrics tracking.
+func (c *ParquetConverter) processBlock(ctx context.Context, userID string, meta *block.Meta, uBucket objstore.InstrumentedBucket, logger log.Logger) {
```

Comment:
general cleanup: the logger is on the `ParquetConverter` struct, we do not need to pass it as an arg to its methods